frontend/src/components/Map/index.js

```diff
@@ -23,617 +32,640 @@ export default function Map() {
-      })
-      .catch(function (error) {
-        console.log(error);
-      });
-
+    const worldData = fetchCovidData(dataScope.WORLD);
+    const usStatesData = fetchCovidData(dataScope.US_STATES);
     map.on("load", function () {
-      // Add source for state polygons hosted on Mapbox, based on US Census Data:
-      // https://www.census.gov/geo/maps-data/data/cbf/cbf_state.html
-      map.addSource("states", {
-        type: "vector",
-        url: "mapbox://mapbox.us_census_states_2015",
-      });
+      addWorldLayer(map, worldData);
+      addUSStatesLayer(map, usStatesData);
+      // addUSCountyLayer(map);
+    });
+  }, []);
 
-      // exclude states outside the 50 states
-      const expression = ["match", ["get", "STATE_ID"]];
-      // var maxValue= data.reduce((max,b) => Math.max(max, b.death), data[0].death);
-      const normalizerValue = 5000;
-      data.forEach(function (row) {
-        var stateID = row.state;
-        if (stateID in stateToFIPS) {
-          var red = (row.death / normalizerValue) * 255;
-          var color = "rgba(" + red + ", " + 0 + ", " + 0 + ", 1)";
-          expression.push(stateToFIPS[stateID], color);
-        }
-      });
+  return (
+    <div className={styles.root}>
+      <div className={classNames(styles.fill)} id="map"></div>
+      {!draggable && <div className={classNames(styles.fill, styles.mask)} />}
+    </div>
+  );
+}
+
+async function fetchCovidData(scope) {
+  return await api(`data/` + scope, {
+    method: "GET",
+  });
+}
 
-      // Last value is the default, used where there is no data
-      expression.push("rgba(0,0,0,0)");
+function addWorldLayer(map, data) {
+  data.then(function (covidData) {
+    const expression = covidData.map(function (row) {
+      return [row.country, get_color(row.group)];
+    });
 
-      const zoomThreshold = 4;
-      // Add layer from the vector tile source with data-driven style
-      map.addLayer(
-        {
-          id: "states-join",
-          type: "fill",
-          source: "states",
-          maxzoom: zoomThreshold,
-          "source-layer": "states",
-          paint: {
-            "fill-color": expression,
+    map.addLayer(
+      {
+        id: "maine",
+        type: "fill",
+        layout: {},
+        maxzoom: countryMinZoom,
+        paint: {
+          "fill-color": {
+            type: "categorical",
+            property: "name",
+            stops: expression,
           },
-        }
-        // 'road-label'
-      );
+          "fill-opacity": 0.7,
+          "fill-outline-color": fillOutlineColor,
+        },
+        source: {
+          type: "vector",
+          url: "mapbox://saurabhp.countries_tileset",
+        },
+        "source-layer": "countries",
+      },
+      "waterway-label"
+    );
+  });
+}
 
-      /*----------------------------  County Level Data ----------------------------------*/
+function get_color(group) {
+  return "rgba(" + group * 255 + ", " + 0 + ", " + 0 + ", 1)";
```

Comment:
*yells about template literals in the background*

Remember also, camelCase is preferred to snake_case in JS! (I know, I know - it's a pain having a snakey python backend)

Reply:
ups, my bad! I knew it, but switching from python to js in the same issue must have glitched my head 🥴